lib/ffaker/number.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+module FFaker
+  module Number
+    extend ModuleUtils
+    extend self
+
+    def number(digits: 1)
+      raise ArgumentError.new('Digits counter should be 1 or greater') if digits < 1
+
+      pattern = '#' * digits
+      result = FFaker.numerify(pattern)
+      return result.to_i if result.chars.first != '0'
+
+      result.tap { |str| str[0] = fetch_sample((1..9).to_a).to_s }.to_i
```

Comment:
It feels like we're relying on chance to generate a "good" number. How do you feel about not leaving that to chance and always generate a good number on the first pass:
```suggestion
      pattern = '#' * (digits - 1)
      leading_digit = rand(1..9)
      FFaker.numerify("#{leading_digit}#{pattern}").to_i
```

You could extract this out to a private method and allow it to take either leading digit safety or trailing digit safety and reuse the logic for `decimal`.

Reply:
Sorry, don't want to sound rude, but what the purpose of such change? I mean.. doesn't it work the same? If it generates "good" number - let it go. if not - replace invalid '0' and that's all.